src/poetry/installation/installer.py

```diff
@@ -281,18 +269,8 @@ def _do_install(self, local_repo: Repository) -> int:
                 # If we are only in lock mode, no need to go any further
                 return 0
 
-        if self._without_groups or self._with_groups or self._only_groups:
-            if self._with_groups:
-                # Default dependencies and opted-in optional dependencies
-                root = self._package.with_dependency_groups(self._with_groups)
-            elif self._without_groups:
-                # Default dependencies without selected groups
-                root = self._package.without_dependency_groups(self._without_groups)
-            else:
-                # Only selected groups
-                root = self._package.with_dependency_groups(
-                    self._only_groups, only=True
-                )
+        if self._groups is not None:
+            root = self._package.with_dependency_groups(self._groups, only=True)
```

Comment:
```suggestion
            root = self._package.with_dependency_groups(list(self._groups), only=True)
```

Reply:
Ack, I was hoping to make the type hint change in core to be `Iterable` than `list`. But we can change that later.

Reply:
Actually, it is not just the type hint but the implementation. The method requires at least a `Container`. You can't do `group_name not in groups` if `groups` is only an `Iterable`.